Python/Product/PythonTools/PythonTools/Options/PythonAnalysisOptions.cs

```diff
@@ -0,0 +1,123 @@
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Linq;
+using Microsoft.Python.Core;
+using Microsoft.PythonTools.LanguageServerClient;
+
+namespace Microsoft.PythonTools.Options {
+    public sealed class PythonAnalysisOptions {
+        private const string Category = "Analysis";
+        private readonly PythonToolsService _service;
+
+        public bool AutoSearchPaths { get; set; }
+        public string DiagnosticMode { get; set; }
+        public string LogLevel { get; set; }
+        public string StubPath { get; set; }
+        public string TypeCheckingMode { get; set; }
+        public string[] TypeshedPaths { get; set; }
+        public bool UseLibraryCodeForTypes { get; set; }
+        public string[] ExtraPaths { get; set; }
+
+        internal PythonAnalysisOptions(PythonToolsService service) {
+            _service = service;
+            Load();
+        }
+
+        public void Load() {
+            var changed = false;
+            var autoSearchPaths = _service.LoadBool(nameof(AutoSearchPaths), Category) ?? true;
+            if(AutoSearchPaths != autoSearchPaths) {
+                AutoSearchPaths = autoSearchPaths;
+                changed = true;
+            }
+
+            var diagnosticMode = _service.LoadString(nameof(DiagnosticMode), Category) ?? PythonLanguageClient.DiagnosticMode.OpenFilesOnly;
+            if(DiagnosticMode != diagnosticMode) {
+                DiagnosticMode = diagnosticMode;
+                changed = true;
+            }
+
+            var logLevel = _service.LoadString(nameof(LogLevel), Category) ?? PythonLanguageClient.LogLevel.Information;
+            if (LogLevel != logLevel) {
+                LogLevel = logLevel;
+                changed = true;
+            }
+
+            var stubPath = _service.LoadString(nameof(StubPath), Category);
+            if (StubPath != stubPath) {
+                StubPath = stubPath;
+                changed = true;
+            }
+
+            var typeCheckingMode = _service.LoadString(nameof(TypeCheckingMode), Category) ?? PythonLanguageClient.TypeCheckingMode.Basic;
+            if (TypeCheckingMode != typeCheckingMode) {
+                TypeCheckingMode = typeCheckingMode;
+                changed = true;
+            }
+
+            var typeshedPaths = _service.LoadString(nameof(TypeshedPaths), Category)?.Split(';');
```

Comment:
If it's a multi-line text box, users may be tempted to put newlines in there. Should we split on `';\r\n'` for extra convenience?

Reply:
oh wrong place

Reply:
Yes, makes sense. I probably also need to add some text to the page wrt 'use line breaks or semicolon to separate paths'. Long term I am going to open task to implement better UI like

![image](https://user-images.githubusercontent.com/12820357/95367938-3e613880-088a-11eb-9dd9-4c514c7b0313.png)

Reply:
Done.